src/lightning_lite/plugins/collectives/torch_.py

```diff
@@ -100,6 +100,8 @@ def scatter_object_list(
         return scatter_object_output_list
 
     def barrier(self, device_ids: Optional[List[int]] = None) -> None:
+        if dist._rank_not_in_group(self.group):
```

Comment:
If we want to avoid protected API:

```suggestion
        if group == dist.GroupMember.NON_GROUP_MEMBER
```

Reply:
+1 for this

Reply:
Please!